First stab at detecting the Safari adopted style sheet bug

DmitrySharabin opened the pull request:
Partially addresses #120

LeaVerou commented on src/util/detect-bugs/adopted-style-sheet.js:
```diff
@@ -0,0 +1,47 @@
+export default {
+	value: undefined,
+	get valuePending () {
+		if (this.value !== undefined) {
```
It seems nothing in here is async? So this should probably be a self-overwriting getter on `value`, with `valuePending` just resolving to that. We can then tweak `detect-bugs` to synchronously resolve if `value` is available (if it doesn't do that already)

DmitrySharabin replied:
> It seems nothing in here is async?

Yup, it's sync. I'll try to refactor it using your proposal. Thanks!

DmitrySharabin commented:
I addressed your feedback. Thank you! The only thing left to do is integrate it into the rest of the code. Other than that, I'm ready for the next iteration.
